packages/react-native-editor/src/test/index.test.js

```diff
@@ -27,158 +29,202 @@ const initGutenberg = ( registerParams ) => {
-		const onRenderEditorCallOrder =
-			onRenderEditor.mock.invocationCallOrder[ 0 ];
-		const hookName = doAction.mock.calls[ hookCallIndex ][ 0 ];
-
-		expect( hookName ).toBe( 'native.pre-render' );
-		expect( hookCallOrder ).toBeLessThan( onRenderEditorCallOrder );
-	} );
+			expect( blockList ).toHaveProperty( 'type', 'View' );
+			expect( console ).toHaveLogged( 'Hermes is: true' );
+		} );
 
-	it( 'dispatches "native.block_editor_props" hook before the editor is rendered', () => {
-		const applyFilters = jest.spyOn( wpHooks, 'applyFilters' );
-
-		// An empty component is provided in order to listen for render calls of the editor component.
-		const onRenderEditor = jest.fn();
-		const EditorComponent = () => {
-			onRenderEditor();
-			return null;
-		};
-		jest.mock( '../setup', () => ( {
-			__esModule: true,
-			default: jest.fn().mockReturnValue( <EditorComponent /> ),
-		} ) );
-
-		initGutenberg();
-
-		const hookCallIndex = 0;
-		// "invocationCallOrder" can be used to compare call orders between different mocks.
-		// Reference: https://git.io/JyBk0
-		const hookCallOrder =
-			applyFilters.mock.invocationCallOrder[ hookCallIndex ];
-		const onRenderEditorCallOrder =
-			onRenderEditor.mock.invocationCallOrder[ 0 ];
-		const hookName = applyFilters.mock.calls[ hookCallIndex ][ 0 ];
-
-		expect( hookName ).toBe( 'native.block_editor_props' );
-		expect( hookCallOrder ).toBeLessThan( onRenderEditorCallOrder );
-	} );
+		it( 'initializes the editor with initial HTML', async () => {
+			const { getByTestId } = initGutenberg(
+				{},
+				{ initialData: initialHtmlGutenberg }
+			);
 
-	it( 'dispatches "native.render" hook after the editor is rendered', () => {
-		const doAction = jest.spyOn( wpHooks, 'doAction' );
-
-		// An empty component is provided in order to listen for render calls of the editor component.
-		const onRenderEditor = jest.fn();
-		const EditorComponent = () => {
-			onRenderEditor();
-			return null;
-		};
-		jest.mock( '../setup', () => ( {
-			__esModule: true,
-			default: jest.fn().mockReturnValue( <EditorComponent /> ),
-		} ) );
-
-		initGutenberg();
-
-		const hookCallIndex = 1;
-		// "invocationCallOrder" can be used to compare call orders between different mocks.
-		// Reference: https://git.io/JyBk0
-		const hookCallOrder =
-			doAction.mock.invocationCallOrder[ hookCallIndex ];
-		const onRenderEditorCallOrder =
-			onRenderEditor.mock.invocationCallOrder[ 0 ];
-		const hookName = doAction.mock.calls[ hookCallIndex ][ 0 ];
-
-		expect( hookName ).toBe( 'native.render' );
-		expect( hookCallOrder ).toBeGreaterThan( onRenderEditorCallOrder );
-	} );
+			const blockList = await waitFor( () =>
+				getByTestId( 'block-list-wrapper' )
+			);
 
-	it( 'initializes the editor', () => {
-		const { getByTestId } = initGutenberg();
-		const blockList = waitFor( () => getByTestId( 'block-list-wrapper' ) );
-		expect( blockList ).toBeDefined();
+			expect( blockList ).toHaveProperty( 'type', 'View' );
```

Comment:
As a workaround for preventing adding false-positives as it was outlined [here](https://github.com/WordPress/gutenberg/pull/37921/files#r783288889), instead of checking the existence, we check that the component type matches what we expect.

Reply:
Interesting technique. 

It is interesting that [RNTL's asynchronous examples](https://callstack.github.io/react-native-testing-library/docs/react-navigation#example-tests) do showcase utilizing `toBeTruthy` when it can easily produce false positives if one forgets to use `await`. 

From reviewing Testing Library's [`findBy` guides](https://testing-library.com/docs/example-findByText), they recommend [leveraging `toBeVisible`](https://github.com/testing-library/jest-dom/tree/v5.16.1#tobevisible), which is a `jest-dom` matcher. Unfortunately, the sibling `jest-native` package does not have a [comparable matcher](https://github.com/testing-library/jest-native/tree/v4.0.4#matchers). 

I wonder if it is worthwhile to [extend `expect`](https://jestjs.io/docs/26.x/expect#expectextendmatchers) our own custom `toBeVisible` matcher to help avoid this common pitfall? WDYT?

Reply:
> It is interesting that RNTL's asynchronous examples do showcase utilizing toBeTruthy when it can easily produce false positives if one forgets to use await.
Yep, I think in general `toBeTruthy` should work for this type of expectation, but I'd lean towards a more robust solution like using a custom expect as `toBeVisible` 👍 .

> From reviewing Testing Library's [`findBy` guides](https://testing-library.com/docs/example-findByText), they recommend [leveraging `toBeVisible`](https://github.com/testing-library/jest-dom/tree/v5.16.1#tobevisible), which is a `jest-dom` matcher. Unfortunately, the sibling `jest-native` package does not have a [comparable matcher](https://github.com/testing-library/jest-native/tree/v4.0.4#matchers).
> 
> I wonder if it is worthwhile to [extend `expect`](https://jestjs.io/docs/26.x/expect#expectextendmatchers) our own custom `toBeVisible` matcher to help avoid this common pitfall? WDYT?

That's very interesting, thanks for sharing it ❤️ . I really like the idea of implementing a matcher like that one, I'll review the implementation of [`toBeVisible`](https://github.com/testing-library/jest-dom/tree/v5.16.1#tobevisible), in case we can provide implementation as close as possible to its original.

Reply:
To be clear, I am happy to merge this assertion as-is. We could always follow up with a `toBeVisible` query at a later time if you'd prefer.

Reply:
> To be clear, I am happy to merge this assertion as-is. We could always follow up with a `toBeVisible` query at a later time if you'd prefer.

Ok, to be honest, it makes sense to me to tackle this in a different PR. Besides, on that PR we could even try to update all tests with that expectation 👍 .

Reply:
@dcalhoun I explored [`@testing-library/jest-native`](https://github.com/testing-library/jest-native) package and [`toBeVisible` matcher](https://github.com/testing-library/jest-dom/blob/main/src/to-be-visible.js) and decided to implement a RN version in [this PR](https://github.com/testing-library/jest-native/pull/73). It's still in draft but I'd appreciate it if you could take a look and let me know your opinion, thanks 🙇 !